### PR TITLE
Fixed a race between a new request and the keepAlive timeout.

### PR DESCRIPTION
**Pull request by maierlars**
Backport of #12194.

https://jenkins.arangodb.biz/job/arangodb-matrix-pr/11059/

### CODE REVIEW

**Comment by maierlars** [comment]
image

File: arangod/GeneralServer/H2CommTask.cpp [hunk 1]
```diff
@@ -0,0 +1,820 @@
+////////////////////////////////////////////////////////////////////////////////
```
Comment (jsteemann) [inline arangod/GeneralServer/H2CommTask.cpp:1]: H2CommTask didn't exist in 3.6 and shouldn't be added there now.

File: arangod/GeneralServer/CommTask.h [hunk 2]
```diff
@@ -87,10 +87,15 @@ class CommTask : public std::enable_shared_from_this<CommTask> {
 
   virtual void start() = 0;
   virtual void close() = 0;
-  
+
+    // returns the number of scheduled requests
+    std::size_t getRequestCount() const { return _requestCount; }
+
+    void setKeepAliveTimeoutReached() { _keepAliveTimeoutReached = true; }
```
Comment (jsteemann) [inline arangod/GeneralServer/CommTask.h:94]: The indentation of the new code doesn't seem to match the indentation of the existing code around it.

**Comment by jsteemann** [comment]
https://172.16.10.101/job/arangodb-matrix-pr/11097/

**Comment by jsteemann** [comment]
Tests blue again :+1:

**Pull request merged**